google/cloud/bigtable/client.py

```diff
@@ -146,6 +150,7 @@ class Client(ClientWithProject):
     _table_data_client = None
     _table_admin_client = None
     _instance_admin_client = None
+    _new_table_data_client = None
```

Comment:
1. We should avoid calling this "new", because someday it will be old
2. We don't need internal references for both the gapic client and the veneer client, since the veneer holds a copy of the gapic. 

So we can probably remove _table_data_client and _new_table_data_client, and just have a `_data_client`, that holds the veneer.

Then in `def table_data_client(self):`, return `self._data_client._gapic_client`